app/controllers/comments_controller.rb

```diff
@@ -31,11 +38,13 @@ def comment_params
 
   def load_post
     @post = Post.find_by id: params[:post_id]
+    flash.alert = t "post_not_found"
     redirect_to root_url unless @post
```

Comment:
chỗ này xử lý vẫn chưa ổn anh nhé, anh xem lại nha :D

Reply:
Chú có thế suggest anh không? anh chưa thấy nó chưa ổn chỗ nào cả. Theo anh tham khảo thì liệu có phải return @post trước nếu nil thì mới redirect không?

Reply:
```suggestion
    @post = Post.find_by id: params[:post_id]
    return uless @post
    flash.alert = t "post_not_found"
    redirect_to root_url
```
cái dưới tương tự nhé anh

Reply:
Hình như chú nhầm thì phải, phải là `return if` chứ nhỉ??

Reply:
chuẩn rồi anh, e nhầm tí :D